themes/theme-default/src/client/components/global/CodeGroup.ts

```diff
@@ -30,6 +31,35 @@ export const CodeGroup = defineComponent({
         tabRefs.value = []
       })
     }
+    const isDark = useDarkMode()
+    const groupRef = ref<HTMLDivElement>()
+    // shiki highlighter color & background
+    onMounted(() => {
+      if (!groupRef.value) return
+      const codeBlock = groupRef.value.querySelector(
+        'div[class*="language-"]',
+      ) as HTMLDivElement
+      if (codeBlock && codeBlock.dataset.highlighter === 'shiki') {
+        const lightColor = codeBlock.style.getPropertyValue('--shiki-light')
+        const darkColor = codeBlock.style.getPropertyValue('--shiki-dark')
+        const lightBg = codeBlock.style.getPropertyValue('--shiki-light-bg')
+        const darkBg = codeBlock.style.getPropertyValue('--shiki-dark-bg')
+        watch(
+          isDark,
+          (val) => {
+            groupRef.value!.style.setProperty(
+              '--c-code-group-tab-bg',
+              val ? darkBg : lightBg,
+            )
+            groupRef.value!.style.setProperty(
+              '--c-code-group-tab-title',
+              val ? darkColor : lightColor,
+            )
+          },
+          { immediate: true },
+        )
+      }
+    })
```

Comment:
Does the code work with prismjs? I did not see any fallback from it.

Reply:
PrismJS is functioning properly and does not require JavaScript to calculate styles.